### PR TITLE
Issue 5413 - Allow only one MemberOf fixup task at a time

### PR DESCRIPTION
**Pull request by mreynolds389**
Description:  only allow one fixup task to run at a time, and improve the task logging

relates: https://github.com/389ds/389-ds-base/issues/5413

### CODE REVIEW

**Review by progier389 — APPROVED** [review]
LGTM

File: ldap/servers/plugins/memberof/memberof.c [hunk 1]
```diff
@@ -421,6 +432,8 @@ memberof_postop_close(Slapi_PBlock *pb __attribute__((unused)))
     slapi_sdn_free(&_pluginDN);
     slapi_destroy_rwlock(config_rwlock);
     config_rwlock = NULL;
+    PR_DestroyLock(fixup_lock);
```
Comment (tbordaz) [inline ldap/servers/plugins/memberof/memberof.c:438]: It is initialized in init fct, I would prefer not to destroy it. Either init it in start (memberof_postop_start) and destroy it in close or simply init it in memberof_postop_init and do not bother to free it.

File: ldap/servers/plugins/memberof/memberof.c [hunk 2]
```diff
@@ -2820,9 +2833,16 @@ memberof_fixup_task_thread(void *arg)
     if (!task) {
         return; /* no task */
     }
+
+    PR_Lock(fixup_lock);
+    fixup_running = PR_TRUE;
```
Comment (tbordaz) [inline ldap/servers/plugins/memberof/memberof.c:2841]: Only 'fixup_running' looks protected by the lock. I would unlock just after the setting of fixup_running to make it clear that the others fixup_* are not protected by the lock.

**Review by tbordaz — APPROVED** [review]
The code looks good to me. Only minor comments

File: ldap/servers/plugins/memberof/memberof.c (hunk 2 above)
Comment (mreynolds389) [reply ldap/servers/plugins/memberof/memberof.c:2841]: only "fixup_running" needs to be protected

File: ldap/servers/plugins/memberof/memberof.c (hunk 1 above)
Comment (mreynolds389) [reply ldap/servers/plugins/memberof/memberof.c:438]: I'll move it to the start function, that way dynamic plugins will work correctly and not break anything.

**Pull request merged**